### PR TITLE
Add SASL db auth support

### PR DESCRIPTION
**Pull request by rendhalver**
Redo of #202
See that PR for details.

### CODE REVIEW

**Comment by benlangfeld** [comment]
@BertrandGouny Could we please get a review on this? An OpenLDAP server without SASL support is a very odd thing indeed.

**Comment by BertrandGouny** [comment]
i'm ok with it but this should be optional and do not break or change any existing db / config if not turned on.

File: image/service/slapd/startup.sh [hunk 1]
```diff
@@ -18,6 +18,10 @@ ulimit -n $LDAP_NOFILE
 chown -R openldap:openldap /var/lib/ldap
 chown -R openldap:openldap /etc/ldap
 chown -R openldap:openldap ${CONTAINER_SERVICE_DIR}/slapd
+chown openldap:openldap /etc/sasldb2
+
+# add sasldb credentials
+echo -n "$LDAP_ADMIN_PASSWORD" | saslpasswd2 -p -u "$LDAP_DOMAIN" -c admin
```
Comment (BertrandGouny) [inline image/service/slapd/startup.sh]: warning LDAP_ADMIN_PASSWORD must be set when boostratping a new db but can be omited or empty later on

File: image/service/slapd/assets/config/bootstrap/ldif/02-security.ldif [hunk 2]
```diff
@@ -5,3 +5,18 @@ delete: olcAccess
 add: olcAccess
 olcAccess: to attrs=userPassword,shadowLastChange by self write by dn="cn=admin,{{ LDAP_BASE_DN }}" write by anonymous auth by * none
 olcAccess: to * by self read by dn="cn=admin,{{ LDAP_BASE_DN }}" write by * none
+
+dn: cn=config
```
Comment (BertrandGouny) [inline image/service/slapd/assets/config/bootstrap/ldif/02-security.ldif]: this should be set only if user is willing to use SASL ?

**Comment by blaggacao** [comment]
@rendhalver Maybe could you attend the comments and do a re-base to see if @BertrandGouny might incline towards merging it into the release-1.4.0 branch? - Much appreciated if this series of events would happen before 1.4.0 is declared new stable.

**Pull request closed without merging**